dev/runtime/atc.ts

```diff
@@ -0,0 +1,68 @@
+import { writeFileSync } from "node:fs"
+import { join } from "node:path"
+import type { CallExpression, SourceFile } from "ts-morph"
+import { Project, SyntaxKind } from "ts-morph"
+import ts from "typescript"
+import { type } from "../../src/type.js"
+import { fromHere } from "./src/fs.js"
+
+export const findArktypeReferenceCalls = (paths: string[]) => {
+    const precompiledDir = fromHere(".temp.prearktype")
+    const project = new Project({})
+    for (const path of paths) {
+        project.addSourceFileAtPathIfExists(path)
+    }
+    for (const file of project.getSourceFiles()) {
+        const filteredStatements = getStatements(file)
+        for (const statement of filteredStatements) {
+            const callExpression = statement.getFirstDescendantByKindOrThrow(
+                SyntaxKind.CallExpression
+            )
+            const functionName = callExpression.getFirstDescendantByKindOrThrow(
+                SyntaxKind.Identifier
+            )
+            const stringifiedAttributes = getStringifiedAttributes({
+                functionName: functionName.getText(),
+                callExpression
+            })
+            callExpression
+                .replaceWithText(stringifiedAttributes)
+                .formatText({ placeOpenBraceOnNewLineForControlBlocks: true })
```

Comment:
If we care about formatting, we should just call prettier after rather than invoking custom formatting, unless it would break otherwise. I'm not sure we do care about formatting though since it's an autogenerated file that is not meant to be modified.